posthog/api/test/batch_exports/test_test.py

```diff
@@ -170,3 +177,129 @@ def test_can_run_s3_test_step_for_destination(client: HttpClient, bucket_name, m
+    """Return a Snowflake configuration dictionary to use in tests."""
+    warehouse = os.getenv("SNOWFLAKE_WAREHOUSE", "warehouse")
+    account = os.getenv("SNOWFLAKE_ACCOUNT", "account")
+    role = os.getenv("SNOWFLAKE_ROLE", "role")
+    username = os.getenv("SNOWFLAKE_USERNAME", "username")
+    password = os.getenv("SNOWFLAKE_PASSWORD", "password")
+    private_key = os.getenv("SNOWFLAKE_PRIVATE_KEY")
+    private_key_passphrase = os.getenv("SNOWFLAKE_PRIVATE_KEY_PASSPHRASE")
+
+    config = {
+        "user": username,
+        "warehouse": warehouse,
+        "account": account,
+        "database": database,
+        "schema": schema,
+        "role": role,
+    }
+    if private_key:
+        config["private_key"] = private_key
+        config["private_key_passphrase"] = private_key_passphrase
+        config["authentication_type"] = "keypair"
+    elif password:
+        config["password"] = password
+        config["authentication_type"] = "password"
+    else:
+        raise ValueError("Either password or private key must be set")
+    return config
+
+
+def test_can_run_snowflake_test_step_for_partial_config(client: HttpClient, snowflake_config, temporal):
+    config = {
+        "role": snowflake_config["role"],
+        "schema": snowflake_config["schema"],
+        "account": snowflake_config["account"],
+        "database": snowflake_config["database"],
+        "warehouse": snowflake_config["warehouse"],
+        "table_name": "events",
+        "user": snowflake_config["user"],
+    }
+    if snowflake_config["authentication_type"] == "keypair":
+        config["private_key"] = snowflake_config["private_key"]
+        config["private_key_passphrase"] = snowflake_config["private_key_passphrase"]
+    elif snowflake_config["authentication_type"] == "password":
+        config["password"] = snowflake_config["password"]
+
+    destination_data = {
+        "type": "Snowflake",
+        "config": config,
+    }
+
+    batch_export_data = {
+        "name": "my-production-snowflake-destination",
+        "destination": destination_data,
+        "interval": "hour",
+    }
+
+    organization = create_organization("Test Org")
+    team = create_team(organization)
+    user = create_user("test@user.com", "Test User", organization)
+    client.force_login(user)
+
+    with start_test_worker(temporal):
+        batch_export = create_batch_export_ok(
+            client,
+            team.pk,
+            batch_export_data,
+        )
+
+        batch_export_data["destination"]["config"] = {"account": "Something", "authentication_type": "password"}
+
+        with unittest.mock.patch(
+            "posthog.temporal.batch_exports.destination_tests.DestinationTest.run_step"
+        ) as run_step_mocked:
+            fake_test_step = SnowflakeEstablishConnectionTestStep()
+            fake_test_step.result = DestinationTestStepResult(status=Status.PASSED, message=None)
+            run_step_mocked.return_value = fake_test_step
```

Comment:
Mocking this so that it works in CI. We are testing we can run the step, not necessarily that the step actually runs (steps have their own tests in a different module).